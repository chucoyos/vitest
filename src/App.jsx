import './App.css'
import { useState } from 'react'

function App() {
	const [text, setText] = useState('Before')
	return (
		<div>
			<h1>Hello Vitest</h1>
			<p>{text}</p>
			<input
				type='text'
				placeholder='name'
				value={text}
				onChange={(e) => setText(e.target.value)}
			/>
		</div>
	)
}

export default App
