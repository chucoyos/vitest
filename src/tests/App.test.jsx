import App from '../App'
import { describe, it, expect } from 'vitest'
import { render, screen, fireEvent, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'

describe('App component', () => {
	it('Renders App component', () => {
		render(<App />)
		screen.debug()
		// This is an explicit assertion
		expect(screen.getByRole('heading').textContent).toBe('Hello Vitest')
		// This is an implicit assertion
		screen.getByText('Hello Vitest')
		screen.getByText(/Hello/)
		// use queryBy when asserting for a missing element
		expect(screen.queryByText(/Javascript/)).toBeNull()
	})
	it('Shows header with text Hello Vitest', () => {
		render(<App />)
		expect(screen.getByRole('heading').textContent).toMatch(/Hello Vitest/i)
	})

	it('Shows examples to get header with text Hello Vitest', () => {
		render(<App />)
		expect(screen.getByRole('heading').textContent).toBe('Hello Vitest')
		expect(screen.getByRole('heading').textContent).toEqual('Hello Vitest')
		expect(screen.getByRole('heading').textContent).not.toEqual(
			'Hello Vitest 2'
		)
		expect(screen.getByRole('heading').textContent).not.toBe('Hello Vitest 2')
	})

	it('Shows examples with snapshot', () => {
		const { container } = render(<App />)
		expect(container).toMatchSnapshot()
		expect(screen.getByRole('heading').textContent).toMatchSnapshot()
		expect(screen.getByRole('heading').textContent).toMatchInlineSnapshot(
			`"Hello Vitest"`
		)
	})
	it('Contains an input field', () => {
		render(<App />)
		expect(screen.getByRole('textbox')).toBeInTheDocument()
		expect(screen.getByPlaceholderText('name')).toBeInTheDocument()
	})
	it('Changes input value', () => {
		render(<App />)
		const input = screen.getByRole('textbox')
		expect(input.value).toBe('Before')
		fireEvent.change(input, { target: { value: 'Javascript' } })
		expect(input.value).toBe('Javascript')
		waitFor(() => expect(screen.getByText('Javascript')).toBeInTheDocument())
		screen.debug()
	})
})
