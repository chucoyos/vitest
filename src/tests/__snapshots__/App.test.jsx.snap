// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > Shows examples with snapshot 1`] = `
<div>
  <div>
    <h1>
      Hello Vitest
    </h1>
    <p>
      Before
    </p>
    <input
      placeholder="name"
      type="text"
      value="Before"
    />
  </div>
</div>
`;

exports[`App component > Shows examples with snapshot 2`] = `"Hello Vitest"`;
